Library/Homebrew/livecheck/strategy.rb

```diff
@@ -57,12 +57,15 @@ def from_symbol(symbol)
       # @param regex_provided [Boolean] whether a regex is provided in the
       #   `livecheck` block
       # @return [Array]
-      def from_url(url, livecheck_strategy: nil, regex_provided: nil)
+      def from_url(url, livecheck_strategy: nil, url_provided: nil, regex_provided: nil, block_provided: nil)
         usable_strategies = strategies.values.select do |strategy|
           if strategy == PageMatch
             # Only treat the `PageMatch` strategy as usable if a regex is
             # present in the `livecheck` block
-            next unless regex_provided
+            next unless regex_provided || block_provided
+          elsif strategy == Sparkle
+            # Skip the `Sparkle` strategy if the URL is not specified explicitly.
+            next unless url_provided
```

Comment:
Yes, it seems like this check isn't necessary anymore since it's explicitly specified in all casks in https://github.com/Homebrew/homebrew-cask/pull/95419 now.